### ISSUE TITLE
Deckfaqs resets after time

### ISSUE THREAD

**Issue by Feelgoodlost** [hulkrelax/deckfaqs#26]
Not sure if you're already aware about this issue but can't see anyone else mentioning it.

So for about three weeks or so there seems to be a problem with Deckfaqs where it will reset after a period of time and return to the main menu (kicks you out of a faq, returns to default state). 

The way I used to use it was to play a game I had on the go, and check the guide I had open every now and then in case I missed things. I could hit the three dots to flick back between the game, also the guide would stay stored and it's position saved even when sleeping/resuming the Deck.

It seems to happen if the guide is left alone for 20 minutes? Or so? But I'm not 100% certain. It also seems to require you to activate the pane whereas before it didn't. What I mean by that is before you could play the game, click the three dots, scroll down a guide, hit the dots and resume play - now you have to hit the dots, choose the window, hit return, scroll, hit dots (it doesn't seem like much but if you're following a guide it makes it less enjoyable). Not sure if it's related to the issue.

Absolutely adore the plugin, my most used by far, so hopefully this is something that can be fixed.

Many thanks!

**Comment by pencilcheck** [comment]
Yea I think it probably has to do with some guides that uses too much memory??

**Comment by Feelgoodlost** [comment]
> Yea I think it probably has to do with some guides that uses too much memory??

Hmm but I have used it consistently for about 6 months without the issue which makes me wonder, including some huge, huge faqs (JRPGs) and guides. Sometimes they didn't load the whole thing (so you would search for a chapter which didn't appear), but that was the only memory related issue I came across. Only recently had it happened, and to try to recreate it I've found the issue on any old thing, seems related to a deck update maybe?

Running the 'stable' release by the way.

**Comment by hulkrelax** [comment]
I haven't investigated yet but I have been able a reproduce some reported crashes only when a game is running. It does possibly seem to be steam deck update related because this did not seem to be an occurrence when I first released the plugin.

**Comment by pencilcheck** [comment]
same, it seems to crash in latest update.

**Comment by hulkrelax** [comment]
I believe I have fixed most if not all crashes in DeckFAQs v1.7.0. Be on the lookout for that release on the main store. If you're brave you could also download it off the testing store once it gets pushed to that

**Comment by Feelgoodlost** [comment]
@hulkrelax I've tried out the testing version to see how it runs, and it still seems to crash (or revert to default) after a period sadly. It does seem better, but it's hard to be sure!

**Comment by hulkrelax** [comment]
What do you mean revert to default? Plugins will go back their normal state for a variety of reasons. When I say "crash" I mean like crashing the whole system. I would not classify the plugin going back to the default page a crash. You can reproduce that at any time by pressing B when on the plugin. Decky Loader basically closes out the plugin.

**Comment by Feelgoodlost** [comment]
> What do you mean revert to default? Plugins will go back their normal state for a variety of reasons. When I say "crash" I mean like crashing the whole system. I would not classify the plugin going back to the default page a crash. You can reproduce that at any time by pressing B when on the plugin. Decky Loader basically closes out the plugin.

Yes sorry, so I guess what I mean is that the plugin goes back to the home menu after a period, whereas before it would stay on the FAQ, and meant you could follow it as long as you didn't fully restart.

**Comment by Feelgoodlost** [comment]
> @hulkrelax I've tried out the testing version to see how it runs, and it still seems to crash (or revert to default) after a period sadly. It does seem better, but it's hard to be sure!

> > What do you mean revert to default? Plugins will go back their normal state for a variety of reasons. When I say "crash" I mean like crashing the whole system. I would not classify the plugin going back to the default page a crash. You can reproduce that at any time by pressing B when on the plugin. Decky Loader basically closes out the plugin.
> 
> Yes sorry, so I guess what I mean is that the plugin goes back to the home menu after a period, whereas before it would stay on the FAQ, and meant you could follow it as long as you didn't fully restart.

For me this behaviour breaks the app, as it's not worth finding the game, then faq, then finding where you were on a faq for it to reset in 10 minutes. Sorry if my original message wasn't very clear.

**Comment by klambchop** [comment]
I’m also having the same issue and finding that it happens sometimes within 5 minutes of temporarily backing out of the FAQ. At this point it kind of defeats the purpose of the quick access and convenience of the tweak and would love to see the issue fixed if possible!

**Comment by pencilcheck** [comment]
1.7.1 seems to stop crashing steam deck and game but the guide still will reset and I have to reopen it again everytime I close and open the side bar.

**Comment by Firebolt700** [comment]
Can also confirm that this is an issue I am experiencing as well. Playing Dragon Quest 8 on Citra as a non-Steam game through Emudeck, I use DeckFAQs to have a walkthrough open but it will reset and go to to the DeckFAQs home page if I wait too long to re-open the plugin menu, then I need to search for the guide again.

**Comment by ninjanerdbgm** [comment]
Can confirm I'm also experiencing this.  Looking at the code (not a TS master by any means) it would appear that what's causing the issue is the app state resetting for some reason.

**Comment by MuradHemani** [comment]
I am having the exact same issue. Hope it gets fixed soon.

**Comment by hulkrelax** [comment]
Sorry I've been on vacation and had some other personal stuff going on for like the last month so I've been not able to address this. Agreed that it is something with the app getting reset. When I have some time, I'm gonna try and see if there is any root cause (I still think it's some Steam thing that I cannot control). I have some ideas to refactor the state management so it persists if I cannot figure out any root cause that can be fixed. I know it sucks in the meantime.

**Comment by ninjanerdbgm** [comment]
Nah man, your plugin is my most-used and has been a godsend for a gamer dad who no longer really has the time to figure out shit on his own lol.  Take your time

**Comment by Feelgoodlost** [comment]
Seconded, take your time, it's been brilliant for me and I hope all is well outside of the project. Thank you again for this tool.

**Comment by Feelgoodlost** [comment]
Just wondering if you ever got to the bottom of the cause @hulkrelax? Hopefully it's not too serious an issue, and I also hope that life is going okay for you.

**Comment by ketsugi** [comment]
Still experiencing this issue with KH1.5 walkthrough
